### ISSUE TITLE
Sanyo VCC-HD5400P PTZ Camera with HDMI

### ISSUE THREAD

**Issue by cc6007** [bitfocus/companion-module-requests#207]
I am after a module for a cctv camera which has a hdmi live view out and can be used for live production.
I cant seem to find any documentation which lists cgi commands but i have started tracking what they are for each function.
Ive got PTZ preset position http commands:
http://[IPADDRESS]/cgi-bin/opecmd.cgi?ope=11&execution=1 (last number being the preset) but i cant seem to have a user name and password to work with generic http requests.

Thanks

**Comment by josephdadams** [comment]
Do you have to log in to the web page of the device before you can fire commands? If so, it might be using session cookies for auth.

**Comment by cc6007** [comment]
Yes, if i try to fire a command in a browser which has previously not been logged into the camera i get a prompt from the browser to enter user and password before the command will fire.

**Comment by josephdadams** [comment]
That sounds like simple http auth. Try formatting your command like this: http://username:password@[IPADDRESS]/cgi-bin/opecmd.cgi?ope=11&execution=1

**Comment by cc6007** [comment]
That works a treat in a browser but i cant seem to get to work in generic http in companion. Ive tried with the user, password and ip in the base url but cant seem to get any buttons to work with post command.

**Comment by josephdadams** [comment]
If you’re typing it into the browser and pressing enter, it’s GET, not POST.

**Comment by cc6007** [comment]
Yeah, ive just been doing some reading and using wireshark and found out its GET, still no luck with the generic http module though. Does it matter if i leave the base URL blank and just but the whole url into the button GET command?

**Comment by josephdadams** [comment]
Yes, you should be able to leave the base URL blank and put the whole URL into the action.

**Comment by cc6007** [comment]
Im still struggling to get the user name and password to pass through with companion. When i inspect the traffic data with wireshark i get a 401 Unauthorised response, even when using http://username:password@[IPADDRESS]

**Comment by josephdadams** [comment]
Can you post some screenshots of the login process in the browser? If it's not simple HTTP auth, you won't be able to pass the login info along with a basic generic HTTP request. You would need a module specifically built to talk to your device. Does the device have any protocol documentation that you can provide?
